client/rpc/rpc_sender.go

```diff
@@ -97,8 +98,7 @@ func (s *Sender) Send(_ context.Context, call client.Call) {
 			// there's visiblity that this is happening. Some of the errors
```

Comment:
This comment says that the the increased visibility of log.Warning was desired here. If that's still true, we should keep it (and maybe relax the requirement that retry.Continue requires a non-nil error); otherwise we should remove this part of the comment.